src/include/atomic.h

```diff
@@ -70,6 +70,17 @@ namespace ceph {
       ceph_spin_unlock(&lock);
       return ret;
     }
+    int cas(T o, T n) {
```

Comment:
name to change compare_and_swap is fine. but the return value(bool) I think it's clearly tell caller I have a successful assign. If we return previous value or new value, it would not convenient for caller? From my side, "cas" actually is a nonblock lock usage, it would be clearness for true/false return.